src/plugins/intel_gpu/src/graph/impls/ocl/broadcast.cpp

```diff
@@ -67,11 +75,8 @@ struct broadcast_impl : typed_primitive_impl_ocl<broadcast> {
                     return true;
                 };
 
-                auto input_rank = data_shape.size();
-                auto output_rank = output_shape.size();
-
-                if (!broadcastable(i_layout, o_layout)) {
-                    data_shape.insert(data_shape.begin(), output_rank - input_rank, 1ul);
+                if (!broadcastable(input_pshape.to_shape(), output_pshape.to_shape())) {
```

Comment:
That seems to work incorrectly:
```
input pshape: [384]
output pshape: [1,3,1,384]
processed input pshape: [384]
final input layout: i32:bfyx:384x1x1x1:nopad
final output layout: i32:bfyx:1x3x1x384:nopad
```

If I remove this broadcastable check, then input shape is correct ([1,1,1,384])